lib/ace/ext/papeeria/papeeria_latex_highlight_rules.coffee

```diff
@@ -0,0 +1,136 @@
+define((require, exports, module) ->
+  'use strict'
+  oop = require('../../lib/oop')
+  TextHighlightRules = require('../../mode/text_highlight_rules').TextHighlightRules
+
+  PapeeriaLatexHighlightRules = ->
+    ###*
+    * correct environment into stack
```

Comment:
so, a good comment looks like this

We maintain a stack of nested LaTeX semantic types (e.g. "document", "section", "list") to be able to provide context for autocompletion and other functions. Stack is constructed by the background highlighter; its elements are then propagated to the editor session <describe here who is responsible for that> and become available through getContext method.

This function adds a new type on the stack.